build.gradle

```diff
@@ -33,45 +34,69 @@ allprojects {
 }
 
 subprojects {
+    apply plugin: 'com.github.ben-manes.versions'
     apply plugin: 'java'
     apply plugin: 'kotlin'
-    apply plugin: 'maven'
 
     sourceCompatibility = JavaVersion.VERSION_1_8
 
     repositories {
         maven {
-            url 'http://dl.bintray.com/kotlin/kotlin-eap-1.1'
+            url "https://plugins.gradle.org/m2/"
         }
         mavenCentral()
         jcenter()
     }
 
+    kotlin { experimental { coroutines 'enable' } }
+
+    dependencyUpdates.resolutionStrategy {
+        componentSelection { rules ->
+            rules.all { ComponentSelection selection ->
+                boolean rejected = ['alpha', 'beta', 'rc', 'cr', 'm'].any { qualifier ->
+                    selection.candidate.version ==~ /(?i).*[.-]${qualifier}[.\d-]*/
+                }
+                if (rejected) {
+                    selection.reject('Release candidate')
+                }
+            }
+        }
+    }
+
     dependencies {
         compile "org.jetbrains.kotlin:kotlin-stdlib:$kotlin_version"
+        compile "io.reactivex.rxjava2:rxkotlin:$rxkotlin_version"
+        compile "org.jetbrains.kotlinx:kotlinx-coroutines-core:$kotlin_coroutine_version"
+        compile "org.jetbrains.kotlinx:kotlinx-coroutines-nio:$kotlin_coroutine_version"
+        compile "org.jetbrains.kotlinx:kotlinx-coroutines-rx2:$kotlin_coroutine_version"
+        compile "org.jetbrains.kotlinx:kotlinx-coroutines-reactive:$kotlin_coroutine_version"
```

Comment:
This seems like more dependencies than we need to be using.  I'll come chat with you about them.

Reply:
👍